tests/Mawosoft.PowerShell.WindowsSearchManager.Tests/mocks/MockCatalogManager.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) 2023 Matthias Wolf, Mawosoft.
+
+namespace Mawosoft.PowerShell.WindowsSearchManager.Tests;
+
+// Exceptions NotImplemented vs NotSupported
+// - NotSupported: the original COM class does not support this member.
+// - NotImplemented: the Mock is missing a implementation for this member
+//
+// Internal members are used to setup mock behavior.
+// Everything is virtual to allow minimal changes per testcase.
```

Comment:
Does that make sense or should we add `virtual` later only to those where it actually turns out to be needed.